src/background.js

```diff
@@ -274,18 +316,101 @@ async function onResultPicked(payload) {
       browser.tabs.create({ url: "https://www.mozilla.org/firefox/new/" });
       break;
   }
+
+  // If we're not restarting, open the survey now.
+  if (!willRestart) {
+    maybeOpenSurvey([tip], "picked");
+  }
+}
+
+/**
+ * We open a user survey web page in three cases:
+ *
+ * (1) Treatment: When the user picks a tip.
+ * (2) Treatment: When the user is shown a tip but they don't pick it.
+ * (3) Control: When the user would have been shown a tip.
+ *
+ * For (1), we open the survey 100% of the time.  For (2) and (3), we open it
+ * only some of the time so that the number of users shown the survey in all
+ * three cases is roughly the same.  Since we expect a pick rate of ~2%, we open
+ * the survey in (2) and (3) only 2% of the time this method is called.
+ *
+ * @param {array} tips
+ *   The tip type(s) that triggered the survey.
+ * @param {string} [action]
+ *   The reason for opening the survey:
+ *     * "picked": The user picked a tip.
+ *     * "ignored": One or more tips were shown in an engagement, but the user
+ *       didn't pick any.
+ *   On the control branch, the action should always be "ignored".
+ */
+async function maybeOpenSurvey(tips, action) {
+  if (openedSurvey) {
+    // Don't open the survey more than once per session.
+    return;
+  }
+
+  // Determine whether we should open the survey.  Tests and QA can set
+  // storage.forceSurvey to an integer value to bypass the randomness logic.
```

Comment:
is this trivial to do for QA? Do they have to run the forceSurvey code from the test in the console? Or can they use the WebExt debugging console to set storage? Maybe the comment could be more specific so we can point them to it.

Reply:
Yes, they can do it either way. I can give them a code snippet to run in the console (which I've been doing for other things), or they can set it using the UI in devtools. Running a snippet might be easier. I have a QA document that I will update for this: https://docs.google.com/document/d/1fcbjFp6YViYCuiM1mry5PPn9p0xhUVPoRSemK9Wguyg/edit#